backend/endpoints/rooms.py

```diff
@@ -435,3 +435,76 @@ async def get_members(org_id: str, room_id: str):
             message="Room members retrieved successfully",
         ),
     )
+
+
+@router.patch(
+    "/org/{org_id}/members/{member_id}/rooms/{room_id}",
+    response_model=ResponseModel,
+    status_code=status.HTTP_201_CREATED,
+    responses={
+        200: {"detail": {"room_id": "room_id"}},
+        424: {"detail": "ZC Core Failed"},
+    },
+ )
+async def update_room(
+    org_id: str, member_id: str, room_id: str, request: UpdateRoomRequest, background_tasks: BackgroundTasks
+ ):
+    """Updates a room between users.
+    Updates a room document to the database collection.
+    Returns the document id and updated info if the room is successfully updated
+    while publishing to the user sidebar in the background
+    Args:
+        org_id (str): A unique identifier of an organisation
+        request: A pydantic schema that defines the room request parameters
+        member_id: A unique identifier of the member updating the room
+    Returns:
+        HTTP_200_OK (room updated): {room_id}
+    Raises
+        HTTP_401_UNAUTHORIZED: not a member of room
+        HTTP_401_UNAUTHORIZED: must be an admin to update room
+        HTTP_424_FAILED_DEPENDENCY: room update unsuccessful
+    """
+    # Initialize Database
+    DB = DataStorage(org_id)
+
+    # Get room from database
+    room = await get_room(org_id=org_id, room_id=room_id)
+    data = request.dict()
+
+    # Get the members of the room, if None assign to empty dictionary
+    members = room.get("room_members") or {}
+    updater = members.get(str(member_id))
+
+    if updater == None:
+        raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="not a member of room",
+    )
+
+    if updater.get("role") != Role.ADMIN:
+        raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="must be an admin to update room.",
+    )
+
+    # Insert members into data request to avoid deletion of all members
+    data['members'] = members
```

Comment:
Why do you need this? A patch update only modifies specified fields. How do mean to avoid deletion of all members?

Reply:
For the MongoDB collection, passing in a Json makes it easier to update. If the line of code is not there, the Json will be used to replace the current on it. 

It's easier to request all and update straight up.

Reply:
I still don't understand your explanation. @nodebe